core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/DomainService.java

```diff
@@ -89,7 +89,8 @@ public List<Domain> findAll(Predicate<Domain> filter) {
 
   @PreAuthorize("hasPermission(#domain, 'DELETE')")
   public void delete(Domain domain) {
-    throw new UnsupportedOperationException("Domain deletion not currently supported.");
+    handlerService.handleDelete(domain);
+    domainRepository.delete(domain);
```

Comment:
Don't we need to worry ensure we maintain referential integrity here? Could this leave a lot of entities (`Stream`, `ProcessInputStream` etc) with references to domains that no longer exist?

Reply:
correct.
Added